src/librustc_resolve/lib.rs

```diff
@@ -3123,391 +3123,405 @@ impl<'a> Resolver<'a> {
                                                 path_str),
                                     );
-                                    return (err, candidates);
-                                },
-                            }
+                                }
+                                return (err, candidates);
+                            },
+                            _ => {
+                                err.span_label(
+                                    span,
+                                    format!("did you mean `{} {{ /* fields */ }}`?",
+                                            path_str),
+                                );
+                                return (err, candidates);
+                            },
                         }
-                        return (err, candidates);
-                    }
-                    (Def::Union(..), _) |
-                    (Def::Variant(..), _) |
-                    (Def::VariantCtor(_, CtorKind::Fictive), _) if ns == ValueNS => {
-                        err.span_label(span, format!("did you mean `{} {{ /* fields */ }}`?",
-                                                     path_str));
-                        return (err, candidates);
                     }
-                    (Def::SelfTy(..), _) if ns == ValueNS => {
-                        err.span_label(span, fallback_label);
-                        err.note("can't use `Self` as a constructor, you must use the \
-                                  implemented struct");
-                        return (err, candidates);
-                    }
-                    (Def::TyAlias(_), _) | (Def::AssociatedTy(..), _) if ns == ValueNS => {
-                        err.note("can't use a type alias as a constructor");
-                        return (err, candidates);
-                    }
-                    _ => {}
+                    return (err, candidates);
+                }
+                (Def::Union(..), _) |
+                (Def::Variant(..), _) |
+                (Def::VariantCtor(_, CtorKind::Fictive), _) if ns == ValueNS => {
+                    err.span_label(span, format!("did you mean `{} {{ /* fields */ }}`?",
+                                                 path_str));
+                    return (err, candidates);
+                }
+                (Def::SelfTy(..), _) if ns == ValueNS => {
+                    err.span_label(span, fallback_label);
+                    err.note("can't use `Self` as a constructor, you must use the \
+                              implemented struct");
+                    return (err, candidates);
                 }
+                (Def::TyAlias(_), _) | (Def::AssociatedTy(..), _) if ns == ValueNS => {
+                    err.note("can't use a type alias as a constructor");
+                    return (err, candidates);
+                }
+                _ => {}
             }
+        }
```

Comment:
to here (2).. seems like a good place to break into a function also.

Reply:
Yeah, I agree with this.